renderlib/FileReaderTIFF.cpp

```diff
@@ -317,21 +321,56 @@ readTiffDimensions(TIFF* tiff, const std::string filepath, VolumeDimensions& dim
 
 // return number of bytes copied to dest
 size_t
-copyAndConvert(uint8_t* dest, const uint8_t* src, size_t numBytes, int srcBitsPerPixel)
+copyDirect(uint8_t* dest, const uint8_t* src, size_t numBytes, int srcBitsPerPixel)
+{
+  memcpy(dest, src, numBytes);
+  return numBytes;
+}
+
+// convert pixels
+// this assumes tight packing of pixels in both buf(source) and dataptr(dest)
+// assumes dest is of format IN_MEMORY_BPP
+// return 1 for successful conversion, 0 on failure (e.g. unacceptable srcBitsPerPixel)
+size_t
+convertChannelData(uint8_t* dest, const uint8_t* src, const VolumeDimensions& dims)
 {
+  // how many pixels in this channel:
+  size_t numPixels = dims.sizeX * dims.sizeY * dims.sizeZ;
+  int srcBitsPerPixel = dims.bitsPerPixel;
+
   // dest bits per pixel is IN_MEMORY_BPP which is currently 16, or 2 bytes
   if (IN_MEMORY_BPP == srcBitsPerPixel) {
-    memcpy(dest, src, numBytes);
-    return numBytes;
+    memcpy(dest, src, numPixels * (srcBitsPerPixel / 8));
```

Comment:
it's clear what you are doing but I'd still prefer to see numPixels * srcBytesPerPixel rather to make it crystal clear. (it's a little pedantic so go with your judgement)